commons-math-legacy/src/main/java/org/apache/commons/math4/legacy/optim/nonlinear/scalar/noderiv/SimplexOptimizer.java

```diff
@@ -403,14 +403,11 @@ private static void keepIfBetter(PointValuePair candidate,
             // List is not fully populated yet.
             for (PointValuePair p : list) {
                 final double[] pPoint = p.getPoint();
-                if (Arrays.equals(pPoint, candidatePoint)) {
-                    // Point was already stored.
-                    return;
-                } else {
+                if (!Arrays.equals(pPoint, candidatePoint)) {
```

Comment:
This loop is a bug. There should not be two return statements in the for loop. The candidate should only be stored if it is not the same as all the values in the list. I have updated master with the fix.

As for the rest of the changes these are all of the type:
```
if () {
   x = ...
   return x
} else {
   x = ...
   return x
}
```
to:
```
if () {
   x = ...
} else {
   x = ...
}
return x
```
I do not see any benefit to this change. It introduces an extra jump condition for the first block which must jump then return, rather than just returning.